src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -1,169 +1,115 @@
-import React, { useCallback, useEffect, useMemo, useState } from "react";
+import React, { useCallback } from "react";
+import classNames from "classnames";
+
+import { User } from "types/User";
 
-import { MessageToTheBandReaction, Reactions } from "utils/reactions";
 import { WithId } from "utils/id";
 
-import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
-import { useReactions } from "hooks/useReactions";
-import { useVenueId } from "hooks/useVenueId";
 
-import {
-  DEFAULT_PARTY_NAME,
-  DEFAULT_PROFILE_IMAGE,
-  RANDOM_AVATARS,
-} from "settings";
+import { UserReactions } from "components/molecules/UserReactions";
 
-import { User } from "types/User";
+import { UserAvatar } from "components/atoms/UserAvatar";
 
-// @debt remove styled-components in favour of using our standard scss patterns
-import * as S from "./UserProfilePicture.styles";
 import "./UserProfilePicture.scss";
 
+// const generateRandomAvatarUrl = (id: string) =>
+//   "/avatars/" +
+//   RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
+
+// export interface AvatarUrlProps {
+//   user?: WithId<User>;
+//   miniAvatars?: boolean;
+// }
+
+// const getAvatarUrl = ({ user, miniAvatars }: AvatarUrlProps): string => {
+//   const { id: userId, anonMode, pictureUrl } = user ?? {};
+//
+//   if (!userId || anonMode) {
+//     return DEFAULT_PROFILE_IMAGE;
+//   }
+//
+//   if (!miniAvatars && pictureUrl) {
+//     return pictureUrl;
+//   }
+//
+//   // @debt how is this intended to be used? Why doesn't it use the profile image? It seems to be something set on venue config..
+//   if (miniAvatars) {
+//     return generateRandomAvatarUrl(userId);
+//   }
+//
+//   return DEFAULT_PROFILE_IMAGE;
+// };
+
 export interface UserProfilePictureProp {
-  user: WithId<User>;
+  user?: WithId<User>;
   isAudioEffectDisabled?: boolean;
+  containerClassName?: string;
+  reactionPosition?: "left" | "right";
+
+  /**
+   * @deprecated Note: This feature is currently disabled.
+   */
   miniAvatars?: boolean;
-  avatarClassName?: string;
-  avatarStyle?: object;
-  containerStyle?: object;
-  reactionPosition?: "right" | "left" | undefined;
 }
 
-// @debt This component should be divided into a few with simpler logic. Also, remove `styled components`
-// @debt the UserAvatar component serves a very similar purpose to this, we should unify them as much as possible
-const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
-  isAudioEffectDisabled = true,
-  miniAvatars = false,
-  avatarClassName = "profile-icon",
-  avatarStyle,
-  containerStyle,
-  reactionPosition,
+export const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
   user,
+  isAudioEffectDisabled = true,
+  containerClassName,
+  reactionPosition = "right",
+  // miniAvatars = false,
```

Comment:
Deleted/comment?

Reply:
There’s a comment in the type definition about it already. Does that suffice?

Reply:
Added `@debt` comments to explain the commented code blocks in https://github.com/sparkletown/sparkle/pull/1323/commits/5f2acbd92b4f96c175bf39ab1ffc0c0bb7d92624